src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java

```diff
@@ -350,4 +344,12 @@ public JsonResult unpublishAndDeleteOnCentral(Long dataSourceId) {
 
         return dataSourceRepository.findAllByCentralIdIsNull().collect(Collectors.toList());
     }
+
+    @Override
+    public boolean isDatasourceNameUnique(String name, Long dataSourceId) {
+
+        return dataSourceRepository.findFirstByNameIgnoreCase(name)
```

Comment:
Base on the function name, one first dataSource by name will be found. Is it possible that we already have a few DS with the same name?

Reply:
I have changed this code a bit, but in general the datanode DB has unique constraint on the name column so only one result can be expected